app/resources/api/v2/plate_purpose_resource.rb

```diff
@@ -73,12 +73,20 @@ def input_plate
         @model.type == 'PlatePurpose::Input'
       end
 
-      # Prevents updating existing plate purposes.
+      # Gets the list of fields which are creatable on a PlatePurpose.
       #
       # @param _context [JSONAPI::Resource::Context] not used
-      # @return [Array<Symbol>] empty array
-      def updatable_fields(_context)
-        []
+      # @return [Array<Symbol>] the list of creatable fields.
+      def self.creatable_fields(_context)
+        %i[name stock_plate cherrypickable_target size asset_shape]
```

Comment:
how did you reach this list? curious.
is _input_ readonly as input plates have to be setup first in the SS config so they can be created by manifests?
most (all?) _stock_ plates are as well though as needed for SS submissions config

Reply:
The missing fields are those that are read only on the API.  If you look at the implementation above, `input_plate` is a boolean that's calculated by the logic `@model.type == 'PlatePurpose::Input'` so setting it to a value has no meaning.  We don't have a setter method for that attribute.  The UUID is the only other missing attribute as this is generated upon creation, so supplying one would have no meaning (unless I'm wrong?  But the existing attribute was marked as `readonly: true` although it turns out that doesn't actually make the attribute read only so I did it this way instead.)

Reply:
I've changed the implementation so that a comment can help explain this

Reply:
Ok that all makes sense